app/models/comment.rb

```diff
@@ -1,2 +1,10 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comment_counter_for_post
+    post = Post.find_by(id: post_id)
+    count = Comment.where(post_id: post.id).count
+    post.update(CommentsCounter: count)
```

Comment:
- [ ] There is an association between the post and the comment, we can use direct posts & increment by 1
`post.increment!(:`comments_counter`)`

Reply:
Thank you so much, Issue resolved.